File: python/java_to_python/05-pgmAbstraccion/src/co/edu/campusucc/poo/TipoElectrico.py
from Vehiculo import Vehiculo
import time


class TipoElectrico(Vehiculo):

    def __init__(self):
        super().__init__()
        self.setTipo("Electrico")

    def startingEngine(self):
        print("🔌 Check Electric System...🔌✅")
        for i in range(2):
            print("¡¡🔋 Init... 🔋!!")
            time.sleep(0.5 - (i * 0.25))
        print("Started...🚙💡")

    def stopingEngine(self):
        print("✅ Check Electric Circuit 🔋")
        for i in range(2):
            print("¡¡⛔ Cutting Electric Circuit...⚡!!")
            time.sleep(0.8 - (i * 0.4))
        print("Stoped...⚡⚡")

    def speedUp(self):
        speedUpString = "speed up in Progress: "
        for i in range(super().getVelocidad()):
            speedUpString += "⏩"

        for i in range(10):
            speedUpString += "⏩"
            print(speedUpString)
            time.sleep(0.5 - (i * 0.05))
        super().addSpeed(10)

File: ruby/pgms_ruby/05-pgmAbstraccion/TipoElectrico.rb
require_relative 'Vehiculo'
require 'time'

class TipoElectrico < Vehiculo
  def initialize
    super
    setTipo("Electrico")
  end

  def startingEngine
    puts "🔌 Check Electric System...🔌✅"
    2.times do |i|
      puts "¡¡🔋 Init... 🔋!!"
      sleep(0.5 - (i * 0.25))
    end
    puts "Started...🚙💡"
  end

  def stopingEngine
    puts "✅ Check Electric Circuit 🔋"
    2.times do |i|
      puts "¡¡⛔ Cutting Electric Circuit...⚡!!"
      sleep(0.8 - (i * 0.4))
    end
    puts "Stoped...⚡⚡"
  end

  def speedUp
    speedUpString = "speed up in Progress: "
    super.getVelocidad.times do
      speedUpString += "⏩"
    end

    10.times do |i|
      speedUpString += "⏩"
      puts speedUpString
      sleep(0.5 - (i * 0.05))
    end
    super.addSpeed(10)
  end
end